bindings/python/setup.py

```diff
@@ -136,36 +140,86 @@ def write_b2_python_config(config):
+            "and invoke b2 to match the expectations and behavior of "
+            "distutils (libtorrent will be built against the invoking "
+            "python, etc; note not all behaviors are currently supported). "
+            "The feature set will match the version found on pypi. "
+            "You can selectively override the auto-configuration in "
+            "this mode with --no-autoconf or --b2-args. For example "
+            "--b2-args=python=x.y or --no-autoconf=python will prevent "
+            "python from being auto-configured. "
+            "Note that --b2-args doesn't currently understand implicit features. "
+            "Be sure to include their names, e.g. --b2-args=variant=debug",
+        ),
+        (
+            "b2-args=",
+            None,
+            "The full argument string to pass to b2. This is parsed with shlex, to "
+            "support arguments with spaces. For example: --b2-args 'variant=debug "
+            '"my-feature=a value with spaces"\'',
+        ),
+        (
+            "no-autoconf=",
+            None,
+            "Space-separated list of b2 arguments that should not be "
+            "auto-configured in distutils mode.",
+        ),
         (
             "libtorrent-link=",
             None,
-            "how to link to libtorrent ('static', 'shared' or 'prebuilt')",
+            "(DEPRECATED; use --b2-args=libtorrent-link=...) ",
         ),
         (
             "boost-link=",
             None,
-            "how to link to boost-python ('static' or 'shared')",
+            "(DEPRECATED; use --b2-args=boost-link=...) "
+        ),
+        ("toolset=", None, "(DEPRECATED; use --b2-args=toolset=...) b2 toolset"),
+        (
+            "pic",
+            None,
+            "(DEPRECATED; use --b2-args=libtorrent-python-pic=on) "
+            "whether to compile with -fPIC",
+        ),
+        (
+            "optimization=",
+            None,
+            "(DEPRECATED; use --b2-args=optimization=...) " "b2 optimization mode",
         ),
-        ("toolset=", None, "b2 toolset"),
-        ("pic", None, "whether to compile with -fPIC"),
-        ("optimization=", None, "b2 optimization mode"),
         (
-            "hash",
+            "cxxstd=",
             None,
-            "use a property hash for the build directory, rather than "
-            "property subdirectories",
+            "(DEPRECATED; use --b2-args=cxxstd=...) "
+            "boost cxxstd value (14, 17, 20, etc.)",
         ),
-        ("cxxstd=", None, "boost cxxstd value (11, 14, 17, etc)"),
     ]
 
     boolean_options = BuildExtBase.boolean_options + ["pic", "hash"]
 
     def initialize_options(self):
 
+        self.config_mode = self.CONFIG_MODE_DISTUTILS
+        self.b2_args = ""
+        self.no_autoconf = ""
+
         self.cxxflags = None
         self.linkflags = None
 
+        # TODO: this is for backwards compatibility
+        # loading these files will be removed in libtorrent-2.0
```

Comment:
Based on the number of `RC_1_2` python build-related tickets in the past few weeks, it might help to reference that this is specifically to pick up configuration from `libtool`, and that all of `libtool` goes away in 2.0